src/librustc/ty/layout.rs

```diff
@@ -2108,142 +1988,179 @@ impl<'a, 'tcx> SizeSkeleton<'tcx> {
     }
 }
 
-/// A pair of a type and its layout. Implements various
-/// type traversal APIs (e.g. recursing into fields).
+/// The details of the layout of a type, alongside the type itself.
+/// Provides various type traversal APIs (e.g. recursing into fields).
```

Comment:
Maybe add a comment that there is no mapping between the type and the CachedLayout because enum variants and the data field of a fat pointer have the `ty` of their "parent" but a different layout?

Reply:
Something along the lines of "Not guaranteed to always be identical to `layout_of(ty)`, as Rust types can't represent all the necessary low-level details such as enum variants and synthetic fields of enums (discriminant / niche) and fat pointers"?